### PR TITLE
Qemu Device type fails to boot if disk type is VIRTIO or SCSI_VIRTIO

### PR DESCRIPTION
**Pull request by lifeeth**
Qemu image has a boot root parameter set to /dev/hda2 causing the boot to fail when the image is run with a disk that is backed by VIRTIO or SCSI_VIRTIO. Changing this to a disk by label path allows for booting in all scenarios. 

This PR adds support for the Kernel modules needed for Google Cloud Platform as well. ( SCSI_VIRTIO support )

Closes https://github.com/resin-os/resinos/issues/248

### CODE REVIEW

File: layers/meta-resin-qemu/recipes-bsp/grub/files/grub.cfg [hunk 1]
```diff
@@ -2,5 +2,5 @@ default=resin
 timeout=3
 
 menuentry 'resin'{
-linux /bzImage root=/dev/hda2 rootwait
+linux /bzImage root=/dev/disk/by-label/resin-rootA rootwait
```
Comment (floion) [inline layers/meta-resin-qemu/recipes-bsp/grub/files/grub.cfg]: hi @lifeeth you could do it like this instead here: https://github.com/resin-os/resin-intel/blob/master/layers/meta-resin-intel/recipes-bsp/grub/files/grub.cfg_internal-dev#L7

Comment (lifeeth) [inline layers/meta-resin-qemu/recipes-bsp/grub/files/grub.cfg]: On it

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 Change log
 -----------
 
+* Update linux-yocto_%.bbappend - To enable support for running the image on Google Cloud [Praneeth]
 * Update tests/autohat - Bring in stability improvements and new tests [Praneeth]
```
Comment (floion) [inline CHANGELOG.md]: The policy we chose is that CHANGELOGS chabges should be in a commit themselves

File: layers/meta-resin-qemu/recipes-bsp/grub/files/grub.cfg (hunk 1 above)
Comment (floion) [inline layers/meta-resin-qemu/recipes-bsp/grub/files/grub.cfg]: Also, I think this changes is worth a CHANGELOG entry too

File: CHANGELOG.md (hunk 2 above)
Comment (lifeeth) [inline CHANGELOG.md]: @floion I added in both places - I can remove it from here

Comment (floion) [inline CHANGELOG.md]: Okay, but there should be a commit with just a CHANGELOG modification

File: layers/meta-resin-qemu/recipes-bsp/grub/files/grub.cfg (hunk 1 above)
Comment (lifeeth) [inline layers/meta-resin-qemu/recipes-bsp/grub/files/grub.cfg]: @floion but I did add Update grub.cfg - Search QEMU root device by label instead of hda/sda/vda paths  - which indicates the same right?

File: CHANGELOG.md (hunk 2 above)
Comment (lifeeth) [inline CHANGELOG.md]: Got it - moved to a standalone commit.

**Comment by lifeeth** [comment]
@resin-jenkins retest this please

**Comment by floion** [comment]
@agherzan added the meta-reisn submodule update as part of this PR so that we can do the root=LABEL=... switch.
Please review

**Comment by lifeeth** [comment]
@resin-jenkins retest this please.
